src/Documents/BulkInsertOperation.ts

```diff
@@ -553,6 +553,28 @@ export class BulkInsertOperation extends BulkInsertOperationBase<object> {
         return { id, metadata, getId };
     }
 
+    public tryStoreSync(entity: object, id: string): boolean;
+    public tryStoreSync(entity: object, id: string, metadata: IMetadataDictionary): boolean;
+    public tryStoreSync(
+        entity: object,
+        id: string,
+        metadata?: IMetadataDictionary): boolean {
+
+        if (this.isFlashNeeded() || this._first) {
```

Comment:
`isFlushNeeded`, no?

Reply:
no, first guarantees that stream was ensured and it was initialized.